cmake/CompilerOptions.cmake

```diff
@@ -111,12 +111,17 @@ function(set_arch_target optvar arch)
         string(REGEX REPLACE "-.*" "" target_model "${target}")
 
         # Use -mcpu for all supported targets _except_ for x86 and Apple arm64, where it should be -march.
-        if("${target}" MATCHES "aarch64-apple-darwin" OR "${target}" MATCHES "arm64-apple-darwin")
-            set(arch_opt "-march=${arch} -mtune=${arch}")
-        elseif(target_model MATCHES "x86|i[3456]86" OR target_model MATCHES "amd64" OR target_model MATCHES "aarch64")
-            set(arch_opt "-march=${arch} -mtune=${arch}")
+        
+        if (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" AND CMAKE_CXX_COMPILER_VERSION LESS 15)
+            set(arch_opt "")
```

Comment:
Ok, that'll _work_ but is certainly not ideal. You can try to detect being on
AArch64 and then use `apple-m1`. However, that also isn't complete, there's an M2 already...

Reply:
1. I can't detect for arch, because building a fat binary happens on x86_64 and also runs into this wall. Afaik there's no CMake builtin way to detect universal builds, in which case only horrible hacks will save us.
2. Indeed, we won't know what the right arch is. I don't think having our own little matrix here is a good idea.

Since it's problem with old AppleClang, with time the people hitting this, which was already near 0, will go down. Only reason really is that I can't upgrade AppleClang in Github runners, so until they move to 15, this needs to be handled. Anyone that cares about optimized builds can always do it themselves.

Reply:
Ok, accepted.